db/index.go

```diff
@@ -321,36 +342,66 @@ func (i *collectionUniqueIndex) newUniqueIndexError(
 	return NewErrCanNotIndexNonUniqueFields(doc.ID().String(), kvs...)
 }
 
+func (i *collectionUniqueIndex) getDocumentsIndexRecord(
+	doc *client.Document,
+) (core.IndexDataStoreKey, []byte, error) {
+	key, err := i.getDocumentsIndexKey(doc)
+	if err != nil {
+		return core.IndexDataStoreKey{}, nil, err
+	}
+	if hasIndexKeyNilField(&key) {
+		key.FieldValues = append(key.FieldValues, []byte(doc.ID().String()))
+		return key, []byte{}, nil
+	} else {
+		return key, []byte(doc.ID().String()), nil
```

Comment:
suggestion: There is a `Bytes()` func on `client.DocID`, no need to cast it twice :)

(same thing a few lines above)

Reply:
it's not used in the code that's why I was hesitant. And it seems to produce different result, than normal "bae-..."

Reply:
If it is weird and we don't want to use it, why should we expect our users to use it?

Best to either make it not weird and use it, or delete it I think.

Reply:
This is actually what I wanted to ask on upcoming sync-up. 

Deleting it wasn't part of the PR.

Reply:
In a separate PR, it might be worth modifying the `Bytes()` function to do what we want it to do if we don't use it for anything else.